wodles/aws/tests/test_aws.py

```diff
@@ -370,3 +417,109 @@ def test_custom_get_creation_date(log_file: dict, expected_date: int, aws_custom
+
+    # Simulate a message with 'ReceiptHandle' key
+    sqs_messages = [
+        {
+            'Body': '{"key": "value"}',
+            'ReceiptHandle': 'example_receipt_handle'
+        }
+    ]
+
+    # Call the extract_message_info method and verify the result
+    extracted_messages = processor.extract_message_info(sqs_messages)
+
+    # Define the expected result based on the sample message
+    expected_result = [{'parsed': 'message', 'handle': 'example_receipt_handle'}]
+
+    # Check that the result is equal to the expected result
+    assert extracted_messages == expected_result
+
+    # Verify that the parse_message method was called with the correct message
+    mock_parse_message.assert_called_once_with({'key': 'value'})
+
+
+def test_parse_message():
+    processor = aws_s3.AWSQueueMessageProcessor()
+    message = {"key": "value"}
+    with pytest.raises(NotImplementedError):
+        processor.parse_message(message)
+
+
+def test_process_jsonl():
+    # Simulate a JSONL file with two JSON objects
+    jsonl_data = b'{"event": "data1"}\n{"event": "data2"}\n'
+    file_wrapper = io.TextIOWrapper(io.BytesIO(jsonl_data), encoding='utf-8')
+    
+    # Call the _process_jsonl function
+    result = aws_s3.AWSSubscriberBucket._process_jsonl(file_wrapper)
+
+    # Verify if the results match the expected output
+    expected_result = [{"event": "data1"}, {"event": "data2"}]
+    assert result == expected_result
+
+
+def test_json_event_generator():
+    data = '{"event": "data1"}{"event": "data2"}'
+    generator = aws_s3.AWSSubscriberBucket._json_event_generator(data)
+    events = list(generator)
+    assert events == [{"event": "data1"}, {"event": "data2"}]
+
+
+def test_remove_none_fields():
+    event = {
+        "field1": "value1",
+        "field2": None,
+        "field3": {
+            "nested_field": None,
+            "nested_field2": "value2"
+        }
+    }
+    expected_event = {
+        "field1": "value1",
+        "field3": {
+            "nested_field2": "value2"
+        }
+    }
+    aws_s3.AWSSubscriberBucket._remove_none_fields(event)
+    assert event == expected_event
+
+
+def test_obtain_logs_from_jsonl_gz_file():
+    # Create a mock object of AWSSubscriberBucket class and configure the expected behavior
+    aws_instance_mock = MagicMock(spec=aws_s3.AWSSubscriberBucket)
+    
+    # Configure the behavior of the obtain_logs method for a JSONL.gz file
+    aws_instance_mock.obtain_logs.return_value = [{'key': 'value', 'source': 'custom'}]
+
+    # Call the obtain_logs method of the mock object with a JSONL.gz file
+    logs = aws_instance_mock.obtain_logs(bucket='mi_bucket', log_path='file.jsonl.gz')
+
+    # Verify that the obtained logs match the expected logs
+    expected_logs = [{'key': 'value', 'source': 'custom'}]
+    assert logs == expected_logs
+
+
+def test_arg_validate_security_lake_auth_params():
+    # Capture the standard output to verify error messages
+    captured_output = io.StringIO()
```

Comment:
Maybe we can do a `fixture` setup/teardown of capturing the standard output, as follow:

```
@pytest.fixture
def capture_stdout():
    # Set up: Redirect stdout to the captured_output
    captured_output = io.StringIO()
    sys.stdout = captured_output

    # This code will run before each test that uses the fixture
    yield captured_output

    # Teardown: Restore the standard output
    sys.stdout = sys.__stdout__
```

Reply:
- Given that the `fixture` would be created for the sole purpose of verifying this specific test at the moment, I believe it is appropriate to leave it as it is. If more tests requiring a `fixture` are introduced in the future, it would be advisable to implement it then. For now, the current approach suffices for the scope of this particular test.